### ISSUE TITLE
too-many-function-args false positive with multiple inheritance

### ISSUE THREAD

**Issue by alvarocaceres** [PyCQA/pylint#1233]
### Steps to reproduce

```
import nonexistent  # pylint doesn't have access to this module's source

class C1(object):
  def f(self):
    pass


class C2(nonexistent.D, C1):
  def f(self):
    super(C2, self).f(1)  # [too-many-function-args], false positive
```
### Current behavior
E: 13, 4: Too many positional arguments for method call (too-many-function-args)

### Expected behavior

No error since nonexistent.D might define f.

### pylint --version output

pylint 1.6.4, 
astroid 1.4.8
Python 3.4.3 (default, Nov 17 2016, 01:08:31) 
[GCC 4.8.4]

**Comment by PCManticore** [comment]
I think this is fine as is, since if ``nonexistent.D`` introduces a method ``f``, that supports an extra parameter, then the class hierarchy is wrong in the first place, since you have a method with the same name, requiring a different number of inputs in order to work properly. If pylint had an actual type system, then the current f would be a ``Callable[]``, while the other ``f`` would then be a ``Callable[Int]``, which is definitely not the same.

**Comment by alvarocaceres** [comment]
Fair point, thanks for looking at it.
